app/components/listing_card_component.rb

```diff
@@ -1,9 +1,11 @@
 # frozen_string_literal: true
 
 class ListingCardComponent < ViewComponent::Base
-  def initialize(listing:, mode: :default, saved_item: nil)
+  def initialize(listing:, mode: :default, saved_item: nil, transaction: nil, current_user: nil)
```

Comment:
You don't need to pass the current user into view components because they have access to it already. 

Bascially `current_user` is a helper method you can access in views and controllers and view component also has access to these helper methods. If you have an issue with just doing `current_user` in the view you might have to do `helpers.current_user`. I can't remember.